src/algo/simple_paths.rs

```diff
@@ -101,14 +102,81 @@ where
     })
 }
 
+type Path<T> = Vec<T>;
+
+/// Compute all possible paths in a directed graph from a given starting node, without repeating
+/// edges. For this algorithm, paths are sequences of edges.
+///
+/// starting_node: Explore paths from this node
+///
+/// (*) without repeating edges; a path can still visit the same node multiple times if there are
+/// multiple edges between nodes. For a regular directed graph (not multigraph), this
+/// means all paths without cycles.
+///
+/// The returned map will contain paths for all explored nodes
+pub fn all_non_repeating_paths_from<G>(
+    graph: G,
+    starting_node: G::NodeId,
+) -> HashMap<G::NodeId, Vec<Path<G::EdgeId>>>
+where
+    G: IntoEdges + GraphProp<EdgeType = Directed>,
+    G::NodeId: Eq + Hash,
+    G::EdgeId: Eq + Hash,
+{
+    let mut path_stack = Vec::<Vec<_>>::new();
+    let mut result = HashMap::<G::NodeId, Vec<Path<_>>>::default();
+    // Map edge id to edge target node id
+    // This extra map needed when we don't have graph[edge_id] lookup
+    let mut edge_targets = HashMap::new();
+
+    for edge in graph.edges(starting_node) {
+        edge_targets.entry(edge.id()).or_insert(edge.target());
+        path_stack.push(vec![edge.id()]);
+    }
+
+    result.entry(starting_node).or_default().push(Vec::new());
+
+    // Using a BFS order, explore all possible paths from the starting point
+    let mut new_path_stack = Vec::new();
```

Comment:
I meant to leave this comment last night as part of my review but I must've messed up somewhere.

Why are you calling this a stack if it's accessed in BFS order?